packages/components/src/internal/components/search/FindDerivativesButton.tsx

```diff
@@ -0,0 +1,156 @@
+import React, {FC, memo, useCallback, useMemo} from 'react';
+
+import { Filter } from '@labkey/api';
+
+import { QueryModel } from '../../../public/QueryModel/QueryModel';
+
+import { AppURL } from '../../url/AppURL';
+
+import { FIND_SAMPLES_BY_FILTER_KEY } from '../../app/constants';
+
+import { formatDateTime } from '../../util/Date';
+
+import { EntityDataType } from '../entities/models';
+
+import { ResponsiveMenuButton } from '../buttons/ResponsiveMenuButton';
+
+import { DataClassDataType, SampleTypeDataType } from '../entities/constants';
+
+import { SCHEMAS } from '../../schemas';
+
+import { getSampleFinderLocalStorageKey, isValidFilterFieldSampleFinder, searchFiltersToJson } from './utils';
+import {FieldFilter, FilterProps} from './models';
+import { SAMPLE_FINDER_SESSION_PREFIX } from './constants';
+import { useAppContext } from '../../AppContext';
+import {DisableableMenuItem} from "../samples/DisableableMenuItem";
+import {DisableableButton} from "../buttons/DisableableButton";
+
+const DISABLED_FIND_DERIVATIVES_MSG = 'Unable to find derivative samples using search filters or filters on multi-valued lookup fields';
+
+export const getFieldFilter = (model: QueryModel, filter: Filter.IFilter): FieldFilter => {
+    const colName = filter.getColumnName();
+    const column = model.getColumn(colName);
+
+    return {
+        fieldKey: column?.fieldKey ?? colName,
+        fieldCaption: column?.caption ?? colName,
+        filter,
+        jsonType: column.isLookup() ? 'string' : column?.jsonType ?? 'string', // deferring to 'string' for lookups since lookup display columns default to text fields
+    } as FieldFilter;
+};
+
+export const getSessionSearchFilterProps = (
+    entityDataType: EntityDataType,
+    model: QueryModel,
+    filters: Filter.IFilter[],
+    baseModel?: QueryModel,
+    baseFilter?: Filter.IFilter[]
+): FilterProps[] => {
+    let fieldFilters = [];
+    // optionally include baseFilter when passed without a baseModel (i.e. apply to the same schemaQuery as the other filters)
+    if (baseFilter && !baseModel) {
+        fieldFilters = fieldFilters.concat(baseFilter.map(filter => getFieldFilter(model, filter)));
+    }
+    // always include viewFilters and user defined filters (filterArray)
+    fieldFilters = fieldFilters.concat(filters.map(filter => getFieldFilter(model, filter)));
+
+    const filterProps = [];
+    if (baseModel && baseFilter) {
+        filterProps.push({
+            schemaQuery: baseModel.schemaQuery,
+            filterArray: [getFieldFilter(baseModel, baseFilter[0])],
+            entityDataType:
```

Comment:
parentEntityDataTypes are passed in from Biologics and LKSM and are not the static const version of DataClassDataType and SampleTypeDataType. 

Also, why is there a need to determine entityDataType if it's passed in as param already?

Reply:
the `entityDataType` prop that is passed in is for the main set of filters on the given grid. In this baseFilter case, that can be different then what we want to use. For example the LKSM Source > Samples grids where the grid filters use the `SampleTypeDataType` but the baseFilter is for the source and uses `DataClassDataType`.

Reply:
ah, I see where it matters not. I'll change it so that the usage needs to pass in the baseEntityDataType so we get the labels right in the SF cards